### ISSUE TITLE
SPIR-V compiler crash with yuzu ASTC decoder

### ISSUE THREAD

**Issue by liamwhite** [IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT#159]
### Checklist [README]

- [X] Device is not undervolted nor overclocked
- [X] Device is using the [latest drivers](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/wiki/Intel-driver-updates-and-changelogs)
- [X] Application is not cracked, modded and use the latest patch

### Application [Required]

yuzu

### Processor / Processor Number [Required]

Core i5 12600K

### Graphic Card [Required]

Arc A770 16GB

### GPU Driver Version [Required]

- [ ] 31.0.101.3491 (*latest*)
- [X] 31.0.101.3490
- [ ] 31.0.101.3430
- [ ] 31.0.101.3276
- [ ] 30.0.101.3268
- [ ] 30.0.101.3259
- [ ] 30.0.101.1743

### Rendering API [Required]

- [X] Vulkan
- [ ] OpenGL
- [ ] DirectX12
- [ ] DirectX11
- [ ] DirectX10
- [ ] DirectX9
- [ ] Not applicable

### Windows Build Number [Required]

- [X] Windows 11 22H2
- [ ] Windows 11 21H1 (*October 2021 Update*)
- [ ] Windows 10 21H2 (*November 2021 Update*)
- [ ] Windows 10 21H1 (*May 2021 Update*)
- [ ] Windows 10 20H2 (*October 2020 Update*)
- [ ] Other (Please specify)

### Intel System Support Utility report

[ssu-info.txt](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/9829105/ssu-info.txt)

### Description and steps to reproduce [Required]

The yuzu Nintendo Switch emulator crashes when loading any application with the Vulkan Arc driver due to a crash in SPIR-V shader compiler. The file was produced with glslangValidator and the Vulkan application has no validation errors.

Integrated graphics drivers are not affected. Mesa is not affected. Other driver families (AMD, Nvidia) are not affected.

Here is a link to the issue report: https://github.com/yuzu-emu/yuzu/issues/9072

Here is a link to the lastest version of yuzu:
https://github.com/yuzu-emu/yuzu-mainline/releases

Here is a link to a zipped homebrew example game to run in the emulator which can trigger the crash:
[simple_triangle.zip](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/9829125/simple_triangle.zip) (free software, compiled from [here](https://github.com/switchbrew/switch-examples/tree/master/graphics/opengl/simple_triangle))

---

I have created a minimal application which reproduces the crash here: https://github.com/liamwhite/astc-crash/

Steps for the minimal application:
1. mkdir build
2. cd build
3. cmake ..
4. Open in VS and build main.cpp
5. Run and it will immediately crash

![arc_crash](https://user-images.githubusercontent.com/9658600/196933469-1a4a74f7-44c9-4cec-a333-1f97f5fb447a.png)

### Crash dumps [Required]

[main.dmp.zip](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/9829183/main.dmp.zip)

**Comment by IntelSupport-Rozilah** [comment]
hi @liamwhite, 
Thank you for reaching out about this issue,  can you specify the published game that you used in Yuzu that made the issue happen so that easy to narrow down the issue?

**Comment by liamwhite** [comment]
Hi @IntelSupport-Rozilah ,

The crash can be reproduced with any game. I have provided an example [homebrew game](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/9829125/simple_triangle.zip) in the issue report which you can use for testing. Ensure you are using the Vulkan backend to reproduce the crash.

**Comment by IntelSupport-Rozilah** [comment]
hi @liamwhite,

Thanks for the reply, we have verified the issue with your app [zipped homebrew game](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/9829125/simple_triangle.zip) and it is reproducible but can you specify as well the published game/app that you used, so that we can escalate to next level?

**Comment by liamwhite** [comment]
@IntelSupport-Rozilah 

I initially tested and reproduced the crash with the game Super Mario 3D World + Bowser's Fury.

For your internal prioritization, however, here is a list of every game I have tested that is affected by this crash:
- Bombslinger
- Hyrule Warriors: Age of Calamity
- Luigi's Mansion 3
- Mario Kart 8 Deluxe
- Nintendo Switch Sports
- Pokémon: Let's Go, Pikachu!
- Splatoon 2
- Super Mario 3D All-Stars
- Super Mario 3D World + Bowser's Fury
- SUPER MARIO ODYSSEY
- Super Smash Bros. Ultimate
- The Legend of Zelda: Breath of the Wild
- The Legend of Zelda: Skyward Sword HD
- Xenoblade Chronicles: Definitive Edition
- Xenoblade Chronicles 2
- Xenoblade Chronicles 3

**Comment by IntelSupport-Rozilah** [comment]
hi @liamwhite ,
thanks for your reply, we will start to verify the published game once it is ready and will revert back to you soon. Stay tuned!

**Comment by IntelSupport-Rozilah** [comment]
hi @liamwhite,
The case has been verified on our side and we are able to reproduce it with our configurations as below:
Processor: ALD-S 
Graphic Driver version: 31.0.101.3490
We will be working on a fix. I want to give you a heads-up that the fix may take 3 to 6 months to be included.

**Comment by liamwhite** [comment]
Okay. I will add a workaround to yuzu for the issue for now, then.

**Comment by goldenx86** [comment]
This issue affects more than just the ASTC compute shader we use.
Basically, all games are prone to random SPIR-V compiler crashes, it's of very high priority for us that this is looked into.

**Comment by Valkirie** [comment]
@IntelSupport-Rozilah hey, just checking if this one is in the pipeline.

**Comment by lucianobraga84** [comment]
Hey, Intel. Every Vulkan game crashes sometime because of this unacceptable problem. ARC is supposed to perform better with new generation APIs like DirectX 12 and VULKAN! 

Can't believe you won't even tell us if it this SPIR-V fix is in the pipeline.

(OpenGL on ARC is bad too, but...who cares?)

**Comment by kunit1** [comment]
Hello @Karen-Intel would you be able to provide an update on this issue? It's a serious bug in the intel Vulkan driver and it's been verified by intel many months ago. Any update or prioritization would be appreciated.

**Comment by liamwhite** [comment]
If I take the original compute shader, and comment out two of the case statements in this switch-case block, it is able to compile the shader.

![image](https://user-images.githubusercontent.com/9658600/233818534-8e5be609-2f00-49a5-b0b1-8c3b360b46a1.png)

In fact, it seems like the compiler cannot cope with having six or more case labels in this particular switch block for whatever reason (doesn't really matter what they are, you just have to have at most five).

I found this by running a testcase reduction, which produced a file so fragile that changing almost anything in it causes the compiler to stop crashing: [astc_decoder.zip](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/11302776/astc_decoder.zip). What's going on? Is the internal state of the compiler getting corrupted?

There haven't been any updates for ~6 months, so I would expect to at least see a report of some progress on the issue.

**Comment by kunit1** [comment]
@Arturo-Intel are you able to help here? Thanks.

**Comment by Karen-Intel** [comment]
Hi all. [EstebanIntel](https://github.com/EstebanIntel) will be providing an update in this case soon

Karen

**Comment by EstebanIntel** [comment]
Hi @liamwhite & @kunit1,

Thank you for reporting this issue. Our priority is to target the most popular games and apps to focus our efforts on providing a high quality, stable experience for the broadest set of users. We will continue to improve our software performance and compatibility throughout 2023 and beyond. While we can’t accommodate your request at this time, please watch this [article](https://www.intel.com/content/www/us/en/support/articles/000094636/graphics.html) on our website for any possible changes to this situation.

**Comment by goldenx86** [comment]
What a joke of a company.
Fix RTX Remix then, you still haven't noticed your driver crashes fornthe same reason there.

**Comment by EstebanIntel** [comment]
Hi @goldenx86,

Have you reported the issues on Portal RTX? I couldn't find any report on it in this platform.

Our driver validation process does include several games. But given the number of PC games that exist, we cannot test every single one for every driver version. This is were you and other users formally reporting an issue is really helpful, so we can be made aware that the issue exists and look into it.

**Comment by goldenx86** [comment]
And a report including a test case, and even pointing where the issue is doesn't merit 5 minutes, even after almost 8 months?
I won't be reporting you anything else, it's clear driver quality is not your concern at all.

**Comment by goldenx86** [comment]
I strongly recommend you re-structure how you deal with driver reports, your competition does way better.

**Comment by lucianobraga84** [comment]
It's simple: it is broken, you know what the problem is, so you fix it. And since RTX Remix will be commonly used, you would be fixing AHEAD.

But "thank you, now we know but we won't fix" is...

Wait, it's Intel.

**Comment by DarrenWarrenV** [comment]
@IntelSupport-Rozilah Can you fix this after what 7-8 months? I am your customer and I demand better support with fixing/resolving driver issues. If not, I'll never buy a laptop with Intel CPU again, also I will tell everyone I know how you completely suck and how lazy you are. I will convince them to choose AMD.

**Comment by BuyMyMojo** [comment]
any update on this?

**Comment by aviot** [comment]
It's 5/11/2023 now, will fix it at the next update?

**Comment by nipkownix** [comment]
Classic Intel.

**Comment by goldenx86** [comment]
The Indie Developer Support team at Intel contacted me directly; the issue is being looked at.

**Comment by peigongdsd** [comment]
Any update on this?

**Comment by sherief** [comment]
> The Indie Developer Support team at Intel contacted me directly; the issue is being looked at.

You're welcome.

**Comment by Karen-Intel** [comment]
Hey guys, just a quick update.

Just to clarify this situation, the help you need is quite clear and we have been working backstage to get traction on this.
We have direct comms with @goldenx86 and working alongside with him and his team already. We're making quite a good progress and I'm positive we will have updates.

> Intel support team hard at work

Also, be sure we appreciate all the comments and we read you, YOU help us improve.  
I'm sure I'll be talking to you all soon (We also moderate the[ Intel Game Dev Support Forum ](https://community.intel.com/t5/Developing-Games-on-Intel/bd-p/developing-games-graphics) so if you are a dev in need of help you're welcome to post your case here or in the forum.

Karen

**Comment by RexSonic** [comment]
I’ll believe you once I see something actually delivered

**Comment by Karen-Intel** [comment]
> I’ll believe you once I see something actually delivered

Fair enough :) Keep you posted

**Comment by B-Reif** [comment]
If someone submits a detailed, reproducible driver error, with an executable example attached, why do you need to know about "the published game/app that you used" to actually fix the problem with your software?

**Comment by BelleNottelling** [comment]
Hello, @Karen-Intel. I have a few questions that I certainly would love some honest answers to, and I suspect everyone else following this thread would also appreciate some answers.

> We have direct comms with @goldenx86 and working alongside with him and his team already.

Was the [comment](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/issues/159#issuecomment-1527618555) by @EstebanIntel simply because of poor internal communication, or was this the actual answer two weeks ago?
Truthfully, your comment appears to be as of a direct result of the comments that were made in the [yuzu progress report](https://yuzu-emu.org/entry/yuzu-progress-report-apr-2023/), where the post rightfully bashes on the way this bug report has been handled by stating things like this:
> This is some kind of twisted joke. For comparison, when you do this with NVIDIA, they hire you.

And showing a screenshot of broken rendering with this caption:
> That’s what happens when we’re forced to remove an entire pipeline stage, Intel

That progress report was posted two days ago. There is zero mention of anybody on Intel's team being in contact with @goldenx86 and the tone and wording of everything on the page suggests at the time the yuzu team had absolutely no reason to believe that even an ounce of effort was being put into the problem.

This to me clearly indicates that having "direct comms" started AFTER the report was posted and made public, which suggest that trying to fix the issue is less that Intel actually wants to fix the issue and more that they just want to deal with the negativity surrounding it.

I can absolutely understand not wanting to focus debugging, engineering, and driver development time to a game that simply doesn't have any a large user-base, but to me this situation feels quite different from those for a few reasons.

1. Yuzu is quite popular. It has 25k stars on Github and nearly 10 million downloads.
    a. The yuzu installer has [9,990,184 downloads](https://tooomm.github.io/github-release-stats/?username=yuzu-emu&repository=liftinstall).
2. Unlike most reports you'll come across, this actually should be really easy to reproduce and debug.
    a. The original posted gave a [GitHub repository](https://github.com/liamwhite/astc-crash/) with source code that could be used to easily reproduce the issue. This means Intel wouldn't have needed to go buy an obscure game then spend potentially hours trying to find a way to easily replicate the issue. Just clone that repository and compile it.
    b. Having readily available access to source code that can be used to reproduce this crash should actually help ease the troubleshooting step on your end. You'd be able to see exactly what the code is doing to cause the crash and even modify the application's behavior if needed. That also opens up the ability to use debugging utilities with the application.
3. It's Vulkan! I know that Arc cards are supposed to struggle with older versions of DX and I know those are a low priority. That's fine and to me the expectation for these GPUs, but Vulkan and DX12 are supposed to be the graphics APIs that Arc GPUs handle well. Why would you *not* want to fix issues an issue that significantly hinders the GPUs capabilities to handle something that it is supposed to be really good at?

I have no ill-will towards you or other's who work at Intel. Creating a graphics card and providing support for it is far from a simple task, but a lot of the driver blunders that we see just feel foolish to an end user and it's hard to understand why they are even such big issues? I can't help but wonder if maybe the driver development team is just not big enough to keep up with the very significant number of issues that currently exist? If that is the case, what does will that mean for future Arc GPUs?

I feel like I keep seeing issues like this where there are tasks and workloads that Arc GPUs should absolutely excel at, but instead they are often incapable of doing those things.
For example:
1. Topaz Gigapixel AI, which people have reported as completely broken and unusable on Intel Arc cards, despite coming in a bundle for a lot of people when they buy the graphics card!
    a. Here's a post of the [Topaz Forums](https://community.topazlabs.com/t/intel-arc-a770-crashes-in-gigapixel-ai/38781) on it
    b. And a few on Reddit: [Post 1](https://www.reddit.com/r/IntelArc/comments/12cdb4h/problem_with_image_ai/), [Post 2](https://www.reddit.com/r/IntelArc/comments/z7ubyx/gigapixel_ai_622_warc_desktop_chugging_issue/)
2. Topaz Video Enhance AI.
    a. Once again, The Arc GPUs should be quite capable when it comes to AI related tasks, but Topaz Video AI is *completely* broken for people. They say that you are working on it with them, but then they also have come back and stated that 
3. Video capturing and encode / decode.
    a. These GPUs have killer encoding and decoding hardware on them, but right now it's hardly functional. Trying to record something using Arc Control is pretty much just a giant waste of time, AV1 encoding causes visual artifacts to be added to the video. There's some kind of encode / decode issue in the diver that's effecting some video editors and from what I can tell also appear to be related to the Topaz Video AI issue.


It's just.. sad. There are a lot of ways where these cards can easily outshine AMDs GPUs, but then they completely struggle to achieve those things because of the broken drivers.

Are these issues lasting so long because there aren't enough people on the team?
Or.. does Intel just not really care about that their GPUs can't even be stable and reliable when trying to perform tasks and take advantage of the features that make them actually stand out compared to the other options?

I'm fully expecting that if you respond to this, I will be told that those additional features are a lower priority than popular games. And I get that. I do. But, there's a problem when a product has selling points that are completely unusable because it's software is broken and it's just not a big priority. And if that is actually the case, then I really do think it would be good to bring on more people to the team.

I apologize for the long-winded comment, but the driver situations are frustrating. I bought an Arc GPU because I want Intel to succeed. I want there to properly be a 3rd competitor to AMD and NVIDIA and I would love to see the team at Intel releasing products that offer great value and some really unique features and selling points. But.. it's hard to envision that future and not feel pessimistic when seemingly baseline functionality is broken in the drivers & being written off as a low-priority.

**Comment by sherief** [comment]
@BelleNottelling I have it on good authority (a birdie told me) that direct comms started after an email thread referencing this tweet started internally at Intel: https://twitter.com/SheriefFYI/status/1656689822759792640

**Comment by BelleNottelling** [comment]
> @BelleNottelling I have it on good authority (a birdie told me) that direct comms started after an email thread referencing this tweet started internally at Intel: https://twitter.com/SheriefFYI/status/1656689822759792640
> 
> Edit: Check the timestamps between the tweet and the mention of the Indie Support Team's contact.

Interesting. Thank you for sharing that insight.
I hope that just means someone important at Intel saw the tweet and realized that there's issues they need to resolve, but at the same time it's hard to not feel like they are just responding to the negativity around it rather than actually being concerned about the quality and stability of the drivers.

**Comment by sherief** [comment]
I share your sentiment.

**Comment by Karen-Intel** [comment]
Hey @BelleNottelling 
A few insights on your questions. Thank you for elaborating on many levels

0. Arturo and I are the current GitHub moderators, that's why you see us very active here and that's why this case got on our plate. With Esteban and the rest of the team, we raise our hands and work together when there's anything needed.

1. We contacted the Yuzu team about 2 weeks ago, which means we were actually working in this prior to the Twitter status. Like many of you, we also work on teams that have lots of things to do on queue, so this -response from the Indie Dev team, which is us- had been cooking for a while now and @goldenx86 would not let me lie. My peer Arturo has been in touch with him.

2. About every existing issue regarding ARC and Intel GFX Products -that has been posted here or in any other forum/, has a case that we make sure to report. However, every case has to be analyzed by different teams that have been affected -like us- for many internal things you may be already aware of + the lifecycle of every bug that is reported on every company like ours.

3. So yes, we need help, that's why sometimes we go back and forth, as we don't have 6 hands (tho I'd like to have them, believe me). Like you, we know the potential our products have to rock, however we are maturing them and it wouldn't be possible without your help. We know this situation won't last forever and like you, we're positive the awkward times will pass before we notice.

4. Now about the evidences we already had, which helped us build a solid case, we needed to focus on the compute shader that is not working as expected. Again, justification is key. We are also developers and that's why our engagement in this case was crucial. We are pretty much bringing another perspective that hope will be helpful and eventful for everyone in this thread.

I think I have covered at least a bit of your concerns, I heartedly hope this brings a little peace of mind to whoever needs it. Again, we are here to help, especially -developers in distress-. Talk to you soon

Sincerely
Karen

**Comment by BelleNottelling** [comment]
Hi, @Karen-Intel 
I appreciate the quick response, but I don't think you've fully answered everything I was wondering about.
Namely:
- If you and Arturo are both the current GitHub moderators and seemingly are working on the same team, what's up with the conflicting message from two weeks ago? Especially if that was roughly the same time you had contacted the yuzu team. It doesn't make very much sense to be contacting their team to work on an issue and then leaving a comment stating "we can’t accommodate your request at this time".
- You mentioned your internal process for tracking bug reports and cited that they have to be analyzed by different teams, does this make up a large portion of the average wait time on a bug fix?
    - Additionally, I'm quite curious about that internal process. It seems that issues reported on here seem to get verified relatively quickly. Do they first get verified and then sent off to a team to be triaged and then an in-dept analysis?
    - Has intel considered providing a bit more insight into this process, to help people get a full picture of what actually happens before a bug report is potentially worked on?
    - In a similar vein to the above topic, has it been considered to provide a bit more info to people who report issues? Right now, most people get a pretty generic response that a fix may take 3-6 months. It's quite vague as to what is actually happening behind the scenes. For example, I think most people would interpret that as confirmation that a bug has been verified and that it will be fixed. Having read some of the issues reported on here, though, it seems like that may not entirely be the case? If that message doesn't actually mean an issue is going to be without a doubt worked on and at least attempted to be fixed, I would probably try to reword it. Being told to have to wait for 3-6 months is not a great feeling for people, and if that's not actually a commitment to resolve the problem they will only end up more frustrated at the end when it isn't fixed.
- You did seemingly confirm that your team is perhaps a bit short-handed when it comes to engineers, but you didn't actually comment on a concern I brought up with that. How will that potentially effect future generations of GPUs? And on the same subject, I think it would be a reasonable concern for people to have that the next generation of Arc GPUs will take priority for driver support and that the first generation will be left largely broken in that case. Needless to say, that would really upset the people who gave Intel a chance and bought the first generation cards. I mean, buying a product, dealing with it's MANY imperfections and at the end being left with something that just doesn't work right will leave a bad taste in anyone's mouth.
- Are there plans to bring on more team members? Not only to help reduce bottlenecks in the process of investigating issues reported and resolving them, but also to hopefully help make the experience a bit better for your team & the customers who are having problems?

Once again, I wish no ill-will to you and the other's on your team. I really do appreciate the quick and seemingly honest response.
Thank you for your time and for responding to my questions

Thank you,
Belle.

**Comment by goldenx86** [comment]
I can confirm what @Karen-Intel is saying, I was contacted by Arturo while working in yuzu's progress report (and I wouldn't change my tone there without results on hand), still waiting an official response. We explained the issue, its ramifications, and our discomfort with the current report system to him. After that, it's closed doors from the dev teams, it's not something I can't comment on. 
Next contact was yesterday, confirming that they managed to replicate all issues and are working on it.

What I can say is that the community as a whole is in a negative mood lately, hardware companies have been more disconnected from reality than ever before, so situations like this make all of us, users and consumers, very frustrated.

NVIDIA lying about the GPU shortage and how much mining affected it, then releasing a product series with negative value.
Asus blowing up users' computers, trying to cover the problem up and not providing any warranty.
Intel reinventing the wheel with Arc's driver, and the decisions taken that you can see here.
The price of entry for Zen4 systems.
Inflation affecting many previously stable markets.

The community is tired, angry. This is the moment to re-establish good will and trust with the hardware community, solve the mistakes of the past. Hopefully fixing this issue helps start a trend on the whole industry.

We're open to provide any more information on the issue and help with any testing required. We know what it feels to be short-staffed.

**Comment by sherief** [comment]
Then I owe an apology over my earlier statement about the reason, though I can still confirm I'm aware of at least two email threads circulating within the Intel graphics group referencing the tweet.

**Comment by peigongdsd** [comment]
Looks like ryujinx is having the same issue too.
[https://github.com/yuzu-emu/yuzu/issues/9072](url)
Maybe they're too busy to look into this issue. Hope fixing the SPIR-V compiler will finally save the both emulators. 

INTEL you'd better understand the feeling of a player who bought your ARC A770 16G only because of the trust in your hardware and your advertisement about how ARC series is good for vulkan gaming only to find that he CANNOT EVEN ENTER AN EMULATED GAME. 

Yuzu and ryujinx both have a large user base. I hope you be really serious about this.

**Comment by Tiancxz** [comment]
It's 5/25/2023 now, will fix it at the next update?i cant wait.I am a decade old fan of Intel, and I hope Intel can fix this bug as soon as possible

**Comment by Tiancxz** [comment]
I am so anxious, intel developers can not add fuel, so down the road are not going to buy the arc. Eveyone says AMD's driver garbage, I did not think intel would be so. I hope the developers of intel will pay attention to this problem

**Comment by EstebanIntel** [comment]
Hi all,

We heard you load and clear, and a fix for this issue is currently been worked on by our driver developers. However, I cannot promise the fix will be available in the next driver update, as it is still in the works. I will post an update when the fixed in merged to a public driver.

**Comment by Tiancxz** [comment]
> EstebanIntel

Dear EstebanIntel
As you know, this problem appeared a long time ago last year and the driver developers never fixed it. Maybe this bug is very hard to fix, I am not accusing them, but I am saying that fixing the bug should take priority over optimizing a certain popular game. Because this kind of vicious bug can cause players to be directly unable to play. So I hope you can tell them to prioritize this kind of problem rather than keep putting it off.
Thank you for your quick reply to our question!

**Comment by BuyMyMojo** [comment]
I’d assume there would be different teams working on optimisation and bug fixes, I am glad to hear they are working on it either wayOn 26 May 2023, at 15:04, Tiancxz ***@***.***> wrote:﻿

EstebanIntel

Dear EstebanIntel
As you know, this problem appeared a long time ago last year and the driver developers never fixed it. Maybe this bug is very hard to fix, I am not accusing them, but I am saying that fixing the bug should take priority over optimizing a certain popular game. Because this kind of vicious bug can cause players to be directly unable to play. So I hope you can tell them to prioritize this kind of problem rather than keep putting it off.
Thank you for your quick reply to our question!

**Comment by EstebanIntel** [comment]
Hi all,

We have just posted driver version [31.0.101.4382](https://www.intel.com/content/www/us/en/download/729157/780387/intel-arc-iris-xe-graphics-beta-windows.html?), which has a possible fix for this issue. Can you all test it out?

**Comment by liamwhite** [comment]
I can confirm that the crash reported in this issue is fixed in 31.0.101.4382. Thank you. Should I close this now or wait for it to hit stable?

**Comment by EstebanIntel** [comment]
Thanks for the quick confirmation @liamwhite!

Regarding our public Beta drivers, they are just as stable as the WHQL ones. The only difference is that the public Beta ones don't go through [Microsoft's Windows Hardware Lab Kit (HLK) testing](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/whql-release-signature), and thus, can't be distributed via Windows Update. But both branches go through the same internal validation process within Intel.

**Comment by liamwhite** [comment]
Closing the issue then!

**Comment by goldenx86** [comment]
Thank you.
Now we need to implement a conversion or skip for Float64 shaders to make the game 100% playable on Intel.

**Comment by goldenx86** [comment]
Link seems to be down at the moment.

**Comment by Tiancxz** [comment]
Thank you for your efforts. This bug has been fixed

**Comment by Tiancxz** [comment]
But it seems that only the issues on ARC 750 or ARC graphics card have been fixed. When I switch to XE, I will crash again

**Comment by rookieplayi** [comment]
what about 10th gen cpus driver ?

**Comment by UziT2004** [comment]
TOTK still crashes in some locations while using Intel(R) Iris(R) Xe Graphics
Will there be a fix for it anytime soon?

**Comment by goldenx86** [comment]
@UziT2004 this fixes the major crashes but there's more work to do from yuzu's part.
A shader compiles invalid SPIR-V code causing crashes on gameplay, we're working on this one ATM.
And Tears of the Kingdom uses Float64 shaders on the intro, and Intel hardware removed support for it, so we need to implement a workaround for that.

Report any remaining yuzu issues to our bug tracker please: https://github.com/yuzu-emu/yuzu/issues/new/choose

**Comment by rookieplayi** [comment]
@goldenx86      will you also fix the crash when entering temple of time in intel gpus ?

**Comment by Jonav9** [comment]
> Hi all,
> 
> We have just posted driver version [31.0.101.4382](https://www.intel.com/content/www/us/en/download/729157/780387/intel-arc-iris-xe-graphics-beta-windows.html?), which has a possible fix for this issue. Can you all test it out?

@EstebanIntel I'm using an integrated Intel® Iris® Xᵉ graphics card, and a 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz processor. Will this update work on my laptop? I've been experiencing screen flickering when no particular software crashes, and I tried using the bug report built-in on the Intel graphics command center. But it doesn't work.  Thank you.

**Comment by Jonav9** [comment]
> @goldenx86 will you also fix the crash when entering temple of time in intel gpus ? btw my intel gpu did not crash on the intro

I thought the fix they were talking about here was going to solve that. It crashes in the same part for me. Have you tried this update?

**Comment by rookieplayi** [comment]
> > @goldenx86 will you also fix the crash when entering temple of time in intel gpus ? btw my intel gpu did not crash on the intro
> 
> I thought the fix they were talking about here was going to solve that. It crashes in the same part for me. Have you tried this update?

my cpu is like an 8th gen so they havent update the drivers for 8th-10th gen cpus

**Comment by peigongdsd** [comment]
> > > @goldenx86 will you also fix the crash when entering temple of time in intel gpus ? btw my intel gpu did not crash on the intro
> > 
> > 
> > I thought the fix they were talking about here was going to solve that. It crashes in the same part for me. Have you tried this update?
> 
> my cpu is like an 8th gen so they havent update the drivers for 8th-10th gen cpus

Maybe you should try Mesa on Linux.

**Comment by rookieplayi** [comment]
> > > > @goldenx86 will you also fix the crash when entering temple of time in intel gpus ? btw my intel gpu did not crash on the intro
> > > 
> > > 
> > > I thought the fix they were talking about here was going to solve that. It crashes in the same part for me. Have you tried this update?
> > 
> > 
> > my cpu is like an 8th gen so they havent update the drivers for 8th-10th gen cpus
> 
> Maybe you should try Mesa on Linux.

i have a laptop bruh

**Comment by rookieplayi** [comment]
> > > > @goldenx86 will you also fix the crash when entering temple of time in intel gpus ? btw my intel gpu did not crash on the intro
> > > 
> > > 
> > > I thought the fix they were talking about here was going to solve that. It crashes in the same part for me. Have you tried this update?
> > 
> > 
> > my cpu is like an 8th gen so they havent update the drivers for 8th-10th gen cpus
> 
> Maybe you should try Mesa on Linux.

linux is the same lol

**Comment by FinnHello** [comment]
Hope this could be solved

**Comment by redbelde** [comment]
so... the problem was fixed in iris xe graphics  or only on Intel arc?

**Comment by Kcaj5** [comment]
> so... the problem was fixed in iris xe graphics or only on Intel arc?

Problem is here, Intel iris xe graphics(

**Comment by EstebanIntel** [comment]
Hi all,

To further clarify, this fix is for **Arc only**, as that is what was reported in the original post of this thread.

If a similar issue to this one is still happening in other Intel graphics products, please create a new issue in this GitHub tracker providing the necessary steps to reproduce it using only legally obtained software and/or homebrew software. Please note that Intel cannot support issues on emulated games, since those cannot be obtained legally.


@Jonav9, driver 31.0.101.4382 is integrated graphics on 11th gen Intel processors or newer plus Intel Arc. So yes, you can install this driver on a i7-1165G7 system. However, as I mentioned above, the fix is for Arc graphics only. And, if you need help with any issue other than what is described in the original post of this thread, please submit a ticket at [Intel Online Service Center](https://supporttickets.intel.com/s/?language=en_US)

**Comment by Whimfoome** [comment]
I think this issue should be reopened because the problem is only partly fixed. It's fixed only for Arc GPU's. Iris Xe still has this problem.

**Comment by Krusl9263** [comment]
> I think this issue should be reopened because the problem is only partly fixed. It's fixed only for Arc GPU's. Iris Xe still has this problem.

Well, seems like ASTC decoder is already working on Iris Xe, and at this time Iris Xe outperforms ARC quite a lot on yuzu when playing TOTK.

**Comment by nyanmisaka** [comment]
The SPIR-V compiler issue only existed on Arc dGPU, it doesn't affect Xe iGPU.
So it makes no sense to keep flooding this closed ticket.

https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/issues/159#issuecomment-1572650451

> this fixes the major crashes but there's more work to do from yuzu's part.
A shader compiles invalid SPIR-V code causing crashes on gameplay, we're working on this one ATM.
> Tears of the Kingdom uses Float64 shaders on the intro, and Intel hardware removed support for it, so we need to implement a workaround.

> Report any remaining yuzu issues to our bug tracker please: https://github.com/yuzu-emu/yuzu/issues/new/choose

The rest of the fixes should be done by the Yuzu emulator devs.

**Comment by bulentpc** [comment]
INTEL ARC A770 **Now playable on Ryujinx emulator** - thanks to the producer.

**Comment by nyanmisaka** [comment]
- https://github.com/yuzu-emu/yuzu/pull/10699
- https://github.com/yuzu-emu/yuzu/pull/10693

These fixed the Vulkan device lost issue.

**Comment by Jonav9** [comment]
> * [shader_recompiler: remove barriers in conditional control flow when device lacks support yuzu-emu/yuzu#10699](https://github.com/yuzu-emu/yuzu/pull/10699)
> * [shader_recompiler: translate f64 to f32 when unsupported on host yuzu-emu/yuzu#10693](https://github.com/yuzu-emu/yuzu/pull/10693)
> 
> These fixed the Vulkan device lost issue.

Thank you, that's right. I was able to pass the "temple of time door" part, that previously would crash. It crashed later after getting the recall ability from Zelda. But I'll start checking mods and seeing what works for me.

**Comment by Clownacy** [comment]
> Please note that Intel cannot support issues on emulated games, since those cannot be obtained legally.

Is no one going to comment on this? Not everyone downloads games from shady websites: legally-owned games can be extracted using a homebrew-capable Nintendo Switch for playing in an emulator. Or is that somehow illegal too? It's pretty rude to throw the entire emulation scene under the bus by suggesting that everyone in it is obtaining games illegally.

**Comment by BelleNottelling** [comment]
> > Please note that Intel cannot support issues on emulated games, since those cannot be obtained legally.
> 
> Is no one going to comment on this? Not everyone downloads games from shady websites: legally-owned games can be extracted using a homebrew-capable Nintendo Switch for playing in an emulator. Or is that somehow illegal too? It's pretty rude to throw the entire emulation scene under the bus by suggesting that everyone in it is obtaining games illegally.

Emulation is often seen as at-best legally dubious by a lot of people.
But even outside of that, Nintendo themselves are very trigger-happy with lawsuits and has many times in the past set the precedence that they do not like people reverse engineering, modifying, or emulating their games / hardware. So yeah, I get the frustration that they won't assist with issues on emulated games, but at the same time if I was Intel I would at the very least be concerned if I was going to be doing anything emulation related, most especially if it was emulation of a Nintendo product.
